src/app/tests/TestNullable.cpp

```diff
@@ -165,24 +169,26 @@ static void TestMove(nlTestSuite * inSuite, void * inContext)
     CtorDtorCounter::ResetCounter();
 
     {
-        auto testSrc = MakeNullable<MovableCtorDtorCounter>(400);
-        Nullable<MovableCtorDtorCounter> testDst(std::move(testSrc));
-        NL_TEST_ASSERT(inSuite, CtorDtorCounter::created == 2 && CtorDtorCounter::destroyed == 1);
+        auto testSrc = MakeNullable<MovableCtorDtorCounter>(400);     // construct
+        Nullable<MovableCtorDtorCounter> testDst(std::move(testSrc)); // move construct
+        NL_TEST_ASSERT(inSuite, CtorDtorCounter::created == 2 && CtorDtorCounter::destroyed == 0);
```

Comment:
Hmm. That's quite significant behavior change between `std::optional` and `Optional`. The former doesn't null out the moved optional/nullable while the latter does. I hope our code doesn't rely on this :). We could override this behavior for our Optional/Nullable but now that we started recommending `std::optional` it's probably better to align to the standard...

Reply:
I think std::optional does the right thing. After a move the guarantee is "valid state, however undefined" so code should not rely that the value is usable after move.